pkg/standalone-utils/contracts/test/MockEulerToken.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+// Functionality required is onDeposit it needs to pull ERC20s out
+// of users wallet and transfer into Euler protocol. Similar
+// to on withdrawl. it needs to pull funds from EulerProtocol
+// and send back to users wallet (the underlying token)
+
+import "../relayer/interfaces/IMockEulerProtocol.sol";
+
+import "@balancer-labs/v2-interfaces/contracts/standalone-utils/IEulerToken.sol";
+import "@balancer-labs/v2-interfaces/contracts/solidity-utils/openzeppelin/IERC20.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/test/TestToken.sol";
+
+
+contract MockEulerToken is IEulerToken, TestToken {
+    // from Euler docs:
+    // in order to invest an asset to earn interest, you need to `deposit` into an eToken
+
+    uint256 public exchangeRateMultiplier;
+
+    // solhint-disable-next-line var-name-mixedcase
+    address public immutable ASSET;
```

Comment:
Can call this `_underlying` or something like that and make it private. The `ASSET` nomenclature is specific to Aave and doesn't conform well to our style.